rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -229,73 +245,55 @@ impl PendingOperation for PendingMessage {
             }
         }
 
-        self.submission_data = Some(Box::new(SubmissionData {
+        self.submission_data = Some(Box::new(TxSubmissionData {
             metadata,
             gas_limit,
         }));
         PendingOperationResult::Success
     }
 
     #[instrument]
-    async fn submit(&mut self) -> PendingOperationResult {
-        make_op_try!(|| self.on_reprepare());
-
+    async fn submit(&mut self) {
         if self.submitted {
             // this message has already been submitted, possibly not by us
-            return PendingOperationResult::Success;
+            return;
         }
 
-        // skip checking `is_ready` here because the definition of ready is it having
-        // been prepared successfully and we don't want to introduce any delay into the
-        // submission process.
-
+        debug!("Getting submission_data");
         let state = self
             .submission_data
-            .take()
+            .clone()
             .expect("Pending message must be prepared before it can be submitted");
 
         // We use the estimated gas limit from the prior call to
         // `process_estimate_costs` to avoid a second gas estimation.
-        let tx_outcome = op_try!(
-            self.ctx
-                .destination_mailbox
-                .process(&self.message, &state.metadata, Some(state.gas_limit))
-                .await,
-            "processing message"
-        );
-
-        op_try!(critical: self.ctx.origin_gas_payment_enforcer.record_tx_outcome(&self.message, tx_outcome.clone()), "recording tx outcome");
-        if tx_outcome.executed {
-            info!(
-                txid=?tx_outcome.transaction_id,
-                "Message successfully processed by transaction"
-            );
-            self.submitted = true;
-            self.reset_attempts();
-            self.next_attempt_after = Some(Instant::now() + CONFIRM_DELAY);
-            PendingOperationResult::Success
-        } else {
-            warn!(
-                txid=?tx_outcome.transaction_id,
-                "Transaction attempting to process message reverted"
-            );
-            self.on_reprepare()
+        let tx_outcome = self
+            .ctx
+            .destination_mailbox
+            .process(&self.message, &state.metadata, Some(state.gas_limit))
+            .await;
+        match tx_outcome {
+            Ok(outcome) => {
+                self.set_submission_outcome(outcome);
+            }
+            Err(e) => {
+                error!(error=?e, "Error when processing message");
```

Comment:
I remember talking about this a bit before but now that I see it, it feels weird for failed txs to now make their way to the `confirm` step even when we know that they were unsuccessful. Curious how you feel

Especially as the `CONFIRM_DELAY` seems to be 10 mins atm (which is probably overkill)